azure-devops/create-prdiff.ps1

```diff
@@ -0,0 +1,25 @@
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+[CmdletBinding(PositionalBinding = $False)]
+Param(
+    [Parameter(Mandatory = $True)]
+    [String]$DiffFile
+)
+
+Start-Process -FilePath 'git' -ArgumentList 'diff' `
+    -NoNewWindow -Wait `
+    -RedirectStandardOutput $DiffFile
+if (0 -ne (Get-Item -LiteralPath $DiffFile).Length) {
+    $msg = @(
+        'The formatting of the files in the repo was not what we expected.'
+        'Please access the diff from format.diff in the build artifacts,'
+        'and apply it with `git apply`.'
+        'Alternatively, you can run the `format` CMake target:'
+        '    cmake --build <builddir> --target format'
+        ''
+    )
+    Write-Host ($msg -join "`n")
+    Write-Host [string](Get-Content -LiteralPath $DiffFile)

```

Comment:
When there's a diff, this prints it out all on one line, see #2681:

```
The formatting of the files in the repo was not what we expected.
Please access the diff from format.diff in the build artifacts,
and apply it with `git apply`.
Alternatively, you can run the `format` CMake target:
    cmake --build <builddir> --target format

[string] diff --git a/stl/inc/bitset b/stl/inc/bitset index 95cd55ee..ad507f5b 100644 --- a/stl/inc/bitset +++ b/stl/inc/bitset @@ -114,7 +114,7 @@ public:          }            size_t _Wpos = 0; -        if (   _Count != 0   ) { +        if (_Count != 0) {              size_t _Bits_used_in_word = 0;              auto _Last                = _Ptr + _Count;              _Ty _This_word            = 0; diff --git a/stl/inc/vector b/stl/inc/vector index e7d7f413..29e592a7 100644 --- a/stl/inc/vector +++ b/stl/inc/vector @@ -31,8 +31,8 @@ public:      using iterator_category = random_access_iterator_tag;      using value_type        = typename _Myvec::value_type;      using difference_type   = typename _Myvec::difference_type; -    using pointer                   = typename _Myvec::const_pointer; -    using reference = const value_type&; +    using pointer           = typename _Myvec::const_pointer; +    using reference         = const value_type&;        using _Tptr = typename _Myvec::pointer;  
ScriptHalted
```

I am not sure whether this is related to CRLFs.

Additionally, I find it curious that the output contains `[string]`.